local-cli/generator-macos/templates/macos/Podfile

```diff
@@ -1,32 +1,28 @@
 require_relative '../node_modules/react-native-macos/scripts/react_native_pods'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
+prepare_react_native_project!
+
 target 'HelloWorld-macOS' do
   platform :macos, '10.15'
   use_native_modules!
+
+  # Flags change depending on the env values.
+  flags = get_default_flags()
+
   use_react_native!(
     :path => '../node_modules/react-native-macos',
-    # to enable hermes on macOS, change `false` to `true` and then install pods
-    :hermes_enabled => false
+    # Hermes is now enabled by default. Disable by setting this flag to false.
+    :hermes_enabled => flags[:hermes_enabled],
+    :fabric_enabled => flags[:fabric_enabled],
+    # Flipper is not compatible w/ macOS
+    :flipper_configuration => FlipperConfiguration.disabled,
+    # An absolute path to your application root.
+    :app_path => "#{Pod::Config.instance.installation_root}/.."
   )
 
-  # Pods specifically for macOS target
-end
-
-target 'HelloWorld-iOS' do
-  platform :ios, '12.4'
-  use_native_modules!
-  use_react_native!(:path => '../node_modules/react-native-macos')
-
-  # Enables Flipper.
-  #
-  # Note that if you have use_frameworks! enabled, Flipper will not work and
-  # you should disable these next few lines.
-  use_flipper!
-
   post_install do |installer|
     react_native_post_install(installer)
     __apply_Xcode_12_5_M1_post_install_workaround(installer)
   end
```

Comment:
Still needed for macOS target to fix this issue:
https://github.com/facebook/react-native/issues/32483

Reply:
Bumping folly in React Native upstream is hard :')